### ISSUE TITLE
How to select two frames with exactly static background for human3.6M dataset ?

### ISSUE THREAD

**Issue by GoldExcalibur** [lschmidtke/shape_templates#2]
Thanks for your work !
I currently randomly select two frames from the same video sequence (with same actor, camera, action, etc.) 
But I found that the background is not static due to the movement of actor. So I wonder are there any constraints
for the selected two frames, such as **time difference**, so that these two frames can share the same bounding box ?

**Comment by lschmidtke** [comment]
Hi! Yes, you need to make sure that the background stays the same. I did this by dividing up the image  according to a (regular) 2d grid. Then I group frames by whether the bounding box of the actor falls within a certain box of that grid.
Note here: the boxes of the grid are somewhat larger than the bounding boxes,  otherwise you end up with not enough frames per group to sample from. 

So in the end you would have several folders per action with frames where the actor is confined within the same box of the grid. And then you can randomly sample from that list of frames to form pairs.

**Comment by GoldExcalibur** [comment]
1) If the image size is h*w, then what is the size of course grid ? 
Since you mention that the boxes of grid are larger than gt bboxes, I guess that the course grid can not be obtained by simply calling function like np.meshgrid. 
2) How to group frames by boxes of grid ? What are the requirements for grid-boxes of frames in the same group ?
3) By the way, I did not find the supplementary material metioned in paper. Will this process be elaborated in the supplementari aterial ?

**Comment by lschmidtke** [comment]
> 1. If the image size is h*w, then what is the size of course grid ?
>    Since you mention that the boxes of grid are larger than gt bboxes, I guess that the course grid can not be obtained by simply calling function like np.meshgrid.
> 2. How to group frames by boxes of grid ? What are the requirements for grid-boxes of frames in the same group ?
> 3. By the way, I did not find the supplementary material metioned in paper. Will this process be elaborated in the supplementari aterial ?

1. I was using boxes of size 200x200 px, if I'm not mistaken the original images are 1000x1000, so roughly a factor of 5. 

2. First, you extract the bounding box coordinates from the metadata for each frame. This gives you two tuples (x, y) specifying the top left and bottom right corners of the bounding box. Now here is what I did:
```
top_left = np.floor(bbox[0] / 200)
bottom_right = np.ceil(bbox[1] / 200)
```
I divide the xy coordinates of both corners by my defined grid size. Next, I create folder names specifying the values x_tl, y_tl, x_br, y_br:
```
raster_pos = ','.join([str(top_left[0]), str(top_left[1]), str(bottom_right[0]), str(bottom_right[1])])
```
And now I can check top left and bottom right coordinates divided by grid size and put them in the corresponding folder.
You don't need to name the folders that way, you can also specify a dict to determine where to put the frames.

3. No, it's not part of the supplementary. That one contains details regarding training and network architectures. If you're interested I can send that to you.

**Comment by GoldExcalibur** [comment]
Thanks for your detailed responses !
 I will try your suggestions again. 
Hope you can provide an official pretrained model if possible.

**Comment by lschmidtke** [comment]
> Thanks for your detailed responses ! I will try your suggestions again. Hope you can provide an official pretrained model if possible.

Working on it :)

**Comment by jojolee123** [comment]
> 

Hello, can you describe how to use human3.6M in more detail? It would be better if you can provide the download link of human3.6M dataset you have made, because I am afraid that the paper cannot be reproduced due to the dataset

**Comment by jojolee123** [comment]
> > 1. If the image size is h*w, then what is the size of course grid ?
> >    Since you mention that the boxes of grid are larger than gt bboxes, I guess that the course grid can not be obtained by simply calling function like np.meshgrid.
> > 2. How to group frames by boxes of grid ? What are the requirements for grid-boxes of frames in the same group ?
> > 3. By the way, I did not find the supplementary material metioned in paper. Will this process be elaborated in the supplementari aterial ?
> 
> 1. I was using boxes of size 200x200 px, if I'm not mistaken the original images are 1000x1000, so roughly a factor of 5.
> 2. First, you extract the bounding box coordinates from the metadata for each frame. This gives you two tuples (x, y) specifying the top left and bottom right corners of the bounding box. Now here is what I did:
> 
> ```
> top_left = np.floor(bbox[0] / 200)
> bottom_right = np.ceil(bbox[1] / 200)
> ```
> 
> I divide the xy coordinates of both corners by my defined grid size. Next, I create folder names specifying the values x_tl, y_tl, x_br, y_br:
> 
> ```
> raster_pos = ','.join([str(top_left[0]), str(top_left[1]), str(bottom_right[0]), str(bottom_right[1])])
> ```
> 
> And now I can check top left and bottom right coordinates divided by grid size and put them in the corresponding folder. You don't need to name the folders that way, you can also specify a dict to determine where to put the frames.
> 
> 3. No, it's not part of the supplementary. That one contains details regarding training and network architectures. If you're interested I can send that to you.
> 
> If you're wondering how to crop the images correctly, I was using sections from this very well written code from here: https://github.com/microsoft/human-pose-estimation.pytorch/blob/master/lib/dataset/JointsDataset.py Specifically, take a look at the `__getitem__` function, where an affine transformation is applied according to center and scale of the person bounding box.



> > 1. If the image size is h*w, then what is the size of course grid ?
> >    Since you mention that the boxes of grid are larger than gt bboxes, I guess that the course grid can not be obtained by simply calling function like np.meshgrid.
> > 2. How to group frames by boxes of grid ? What are the requirements for grid-boxes of frames in the same group ?
> > 3. By the way, I did not find the supplementary material metioned in paper. Will this process be elaborated in the supplementari aterial ?
> 
> 1. I was using boxes of size 200x200 px, if I'm not mistaken the original images are 1000x1000, so roughly a factor of 5.
> 2. First, you extract the bounding box coordinates from the metadata for each frame. This gives you two tuples (x, y) specifying the top left and bottom right corners of the bounding box. Now here is what I did:
> 
> ```
> top_left = np.floor(bbox[0] / 200)
> bottom_right = np.ceil(bbox[1] / 200)
> ```
> 
> I divide the xy coordinates of both corners by my defined grid size. Next, I create folder names specifying the values x_tl, y_tl, x_br, y_br:
> 
> ```
> raster_pos = ','.join([str(top_left[0]), str(top_left[1]), str(bottom_right[0]), str(bottom_right[1])])
> ```
> 
> And now I can check top left and bottom right coordinates divided by grid size and put them in the corresponding folder. You don't need to name the folders that way, you can also specify a dict to determine where to put the frames.
> 
> 3. No, it's not part of the supplementary. That one contains details regarding training and network architectures. If you're interested I can send that to you.
> 
> If you're wondering how to crop the images correctly, I was using sections from this very well written code from here: https://github.com/microsoft/human-pose-estimation.pytorch/blob/master/lib/dataset/JointsDataset.py Specifically, take a look at the `__getitem__` function, where an affine transformation is applied according to center and scale of the person bounding box.

1 Hello, can you describe how to use human3.6M in more detail? It would be better if you can provide the download link of human3.6M dataset you have made, because I am afraid that the paper cannot be reproduced due to the dataset
2 Could you also provide me with a supplementary material? lijiaojiao@hust.edu.cn
Thank you very much!

**Comment by jojolee123** [comment]
1 Hello, can you describe how to use human3.6M in more detail? It would be better if you can provide the download link of human3.6M dataset you have made, because I am afraid that the paper cannot be reproduced due to the dataset
2 Could you also provide me with a supplementary material? [lijiaojiao@hust.edu.cn](mailto:lijiaojiao@hust.edu.cn)
Thank you very much!

**Comment by jojolee123** [comment]
> > 1. If the image size is h*w, then what is the size of course grid ?
> >    Since you mention that the boxes of grid are larger than gt bboxes, I guess that the course grid can not be obtained by simply calling function like np.meshgrid.
> > 2. How to group frames by boxes of grid ? What are the requirements for grid-boxes of frames in the same group ?
> > 3. By the way, I did not find the supplementary material metioned in paper. Will this process be elaborated in the supplementari aterial ?
> 
> 1. I was using boxes of size 200x200 px, if I'm not mistaken the original images are 1000x1000, so roughly a factor of 5.
> 2. First, you extract the bounding box coordinates from the metadata for each frame. This gives you two tuples (x, y) specifying the top left and bottom right corners of the bounding box. Now here is what I did:
> 
> ```
> top_left = np.floor(bbox[0] / 200)
> bottom_right = np.ceil(bbox[1] / 200)
> ```
> 
> I divide the xy coordinates of both corners by my defined grid size. Next, I create folder names specifying the values x_tl, y_tl, x_br, y_br:
> 
> ```
> raster_pos = ','.join([str(top_left[0]), str(top_left[1]), str(bottom_right[0]), str(bottom_right[1])])
> ```
> 
> And now I can check top left and bottom right coordinates divided by grid size and put them in the corresponding folder. You don't need to name the folders that way, you can also specify a dict to determine where to put the frames.
> 
> 3. No, it's not part of the supplementary. That one contains details regarding training and network architectures. If you're interested I can send that to you.
> 
> If you're wondering how to crop the images correctly, I was using sections from this very well written code from here: https://github.com/microsoft/human-pose-estimation.pytorch/blob/master/lib/dataset/JointsDataset.py Specifically, take a look at the `__getitem__` function, where an affine transformation is applied according to center and scale of the person bounding box.

1 Hello, can you describe how to use human3.6M in more detail? It would be better if you can provide the download link of human3.6M dataset you have made, because I am afraid that the paper cannot be reproduced due to the dataset
2 Could you also provide me with a supplementary material? [lijiaojiao@hust.edu.cn](mailto:lijiaojiao@hust.edu.cn)
Thank you very much!

**Comment by lschmidtke** [comment]
Hi,

0: We only use bounding boxes and no further annotations in form of landmark labels. 

1: I cannot provide a download link for the dataset because I have no rights to distribute it. You can register and
    download the dataset here: http://vision.imar.ro/human3.6m/description.php We have provided checkpoints of the trained  
    model. If you need more help with preprocessing of the dataset and the answers provided in the issues section are not 
   enough, let me know and I can share some code.

2: The supplementary material only contains accompanying videos.
